### ISSUE TITLE
Zoltan2_AlgScotch.hpp:649:9: error: 'SCOTCH_STRATLEVELMAX' was not declared in this scope

### ISSUE THREAD

**Issue by nschloe** [trilinos/Trilinos#160]
As of recently, I'm getting the config error
```
/«PKGBUILDDIR»/packages/zoltan2/src/algorithms/partition/Zoltan2_AlgScotch.hpp:649:9: error: 'SCOTCH_STRATLEVELMAX' was not declared in this scope
         SCOTCH_STRATLEVELMAX | SCOTCH_STRATLEVELMIN | SCOTCH_STRATLEAFSIMPLE | SCOTCH_STRATSEPASIMPLE,
         ^
```
Full log [here](https://launchpadlibrarian.net/243538118/buildlog_ubuntu-xenial-amd64.trilinos_12.7~20160228030214-xenial1_BUILDING.txt.gz).

**Comment by jwillenbring** [comment]
@kddevin feel free to reassign, but I wanted to make sure someone saw this.

**Comment by kddevin** [comment]
@bddavid 
Brad:  would you please look at this issue?  Let me know how I can help. Thanks.

**Comment by kddevin** [comment]
@nschloe @trilinos/zoltan2 
Nico:  Are you using Scotch version 6?  
Zoltan2:  We probably need to be explicit about checking Scotch version numbers.

**Comment by nschloe** [comment]
This is with libptscotch-5.1, [the latest version available from Debian](https://packages.debian.org/sid/libptscotch-dev).

**Comment by egboman** [comment]
> Zoltan2: We probably need to be explicit about checking Scotch version numbers. 

First we should document which versions of TPL (e.g. Scotch) Zoltan2 supports. Is this already done at the Trilinos level?

**Comment by tawiesn** [comment]
Each package documents itself which TPL versions are supported (see e.g. https://trilinos.org/packages/amesos/ or https://trilinos.org/packages/amesos2/). Most often this information is outdated. A central up-to-date place where all supported TPL versions are documented would be really nice to have.

**Comment by rppawlo** [comment]
Ideally this should be built into Tribits

**Comment by srajama1** [comment]
The best place to handle this is FindTPL****.cmake. See FindTPLSuperLUDist.cmake for an example.

**Comment by kddevin** [comment]
Zoltan2 requires ParMETIS 4.0.3 or later and Scotch 6.0.3 or later.
We should encode this requirement in the FindTPL files as Siva recommended.
Note that Zoltan can work with earlier versions of these TPLs.

**Comment by kddevin** [comment]
I will check with @MicheldeMessieres and @krcb about their availability to address this issue.

**Comment by MicheldeMessieres** [comment]
@kddevin
I’ve modified FindTPLScotch and FindTPLParMETIS to check for Zoltan2 required versions.
PR #661

Scotch requires: 6.0.3
ParMETIS requires: 4.0.3

There was already a compile time check for ParMETIS in TPLTraits.cpp which I removed since it would be redundant to the new config time check.

Also there was a compile time check for MPI and ParMETIS which I changed to a config time check.
Let me know if that is not desirable but it seems more appropriate to have them all together.

Proper behavior was checked using the following versions as tests:
Scotch: 6.0.4 and 5.1.12
ParMETIS: 4.0.3 and 3.2.0
